### DISCUSSION TITLE
About controll checkbox with code

### DISCUSSION THREAD

**Discussion by GuosXin** [GeekyAnts/NativeBase#5256] [General]
### Description

I need to use code to control whether the checkbox is selected or not, I can't do it now.

### Problem Statement

Now I have a select all checkbox. I need to judge whether the user has selected all items, and then I want to control the checkbox to select or not with code. I set the defaultIsChecked prop of the checkbox by use state hook, but it didn't refresh in view. I'm not sure if I wrote it wrong, if I did, please tell me how to make it.

### Proposed Solution or API

I have no idea

**Comment by ankit-tailor** [comment]
Hey @GuosXin, Needs more discussion here. moving this to discussions.

**Reply by GuosXin** [reply]
Hi, can it be like the official checkbox, If I set the value attribute to true, will the checkbox be checked by itself. 
Here is the official link:
[https://reactnative.dev/docs/0.66/checkbox](url)
[https://github.com/react-native-checkbox/react-native-checkbox](url)

**Comment by ehte92** [comment]
If I am putting multiple checkbox insider a checkbox group, then is it possible to clear all the checked boxes by the button click, for example like a clear button? In web, we could do that with the ref and using the clear function, but that doesn't seem to work here.
